agent/connect/ca/provider_consul.go

```diff
@@ -658,3 +660,15 @@ func (c *ConsulProvider) generateCA(privateKey string, sn uint64) (string, error
 
 	return buf.String(), nil
 }
+
+// SetLogger implements the NeedsLogger interface so the provider can log important messages.
+func (c *ConsulProvider) SetLogger(logger *log.Logger) {
+	c.logger = logger
+}
+
+// Printf sends output to the configured Logger, if one exists.
+func (c *ConsulProvider) Printf(format string, v ...interface{}) {
```

Comment:
This appears to not be used.

Reply:
You're right, the Consul provider doesn't really do any logging currently. What happened was that I added a bunch of debugging statements while I was testing CA providers, and then removed the debug code when I was done.

Reply:
Fixed by 7bf3e91d5a5bd15ef8e304703ac57c97cff9a53b.